### PR TITLE
Manage required PHP version according to Symfony version

### PR DESCRIPTION
**Pull request by pgrimaud**

### CODE REVIEW

File: bin/requirements-checker.php [hunk 1]
```diff
@@ -22,7 +22,14 @@
 
 $lineSize = 70;
 $isVerbose = in_array('-v', $argv) || in_array('-vv', $argv) || in_array('-vvv', $argv);
-$symfonyRequirements = new SymfonyRequirements(dirname(dirname(realpath($autoloader))));
+
+if (class_exists('\Symfony\Component\HttpKernel\Kernel') && defined('\Symfony\Component\HttpKernel\Kernel::VERSION')) {
```
Comment (fabpot) [inline bin/requirements-checker.php]: `VERSION` is always defined, so the check can be removed IMHO.

Comment (fabpot) [reply bin/requirements-checker.php]: So, the if condition could be replaced with:

```php
$symfonyVersion = class_exists('\Symfony\Component\HttpKernel\Kernel') ? \Symfony\Component\HttpKernel\Kernel::VERSION : null
```

File: public/check.php [hunk 2]
```diff
@@ -31,7 +31,13 @@
     throw new \RuntimeException('Unable to find the Composer autoloader.');
 }
 
-$symfonyRequirements = new SymfonyRequirements(dirname(dirname(realpath($autoloader))));
+if (class_exists('\Symfony\Component\HttpKernel\Kernel') && defined('\Symfony\Component\HttpKernel\Kernel::VERSION')) {
+    $symfonyVersion = \Symfony\Component\HttpKernel\Kernel::VERSION;
+} else {
+    $symfonyVersion = null;
+}
```
Comment (fabpot) [inline public/check.php]: same as above

File: src/SymfonyRequirements.php [hunk 3]
```diff
@@ -20,9 +20,10 @@
  */
 class SymfonyRequirements extends RequirementCollection
 {
-    const REQUIRED_PHP_VERSION = '5.5.9';
+    const REQUIRED_PHP_VERSION_BELLOW_SF4 = '5.5.9';
```
Comment (fabpot) [inline src/SymfonyRequirements.php]: typo: BELOW

Comment (fabpot) [reply src/SymfonyRequirements.php]: What about `REQUIRED_PHP_VERSION_3x` and `REQUIRED_PHP_VERSION_4x` (I know one can use this lib on 2.x, but as this is an internal detail, I think we're good.)

File: bin/requirements-checker.php (hunk 1 above)
Comment (pgrimaud) [reply bin/requirements-checker.php]: Right, I fixed it

File: public/check.php (hunk 2 above)
Comment (pgrimaud) [reply public/check.php]: Fixed too

File: src/SymfonyRequirements.php (hunk 3 above)
Comment (pgrimaud) [reply src/SymfonyRequirements.php]: Good suggest for `REQUIRED_PHP_VERSION_3x` and `REQUIRED_PHP_VERSION_4x`, I renamed the two constants.

**Comment by fabpot** [comment]
Thank you @pgrimaud.

**Pull request merged**